File: problems/0008_string-to-integer-atoi/one-loop/Solution.cs
public class Solution {
    public int MyAtoi(string str) {
        if (str.Length == 0)
            return 0;
        var i = 0;
        for (; i < str.Length && str[i] == ' '; ++i);
        var sign = 1L;
        if (i < str.Length)
            if (str[i] == '-') {
                i++;
                sign = -1L;
            } else if (str[i] == '+')
                i++;
        var result = 0L;
        for (; i < str.Length && '0' <= str[i] && str[i] <= '9'; ++i) {
            result = result * 10 + sign * (str[i] - '0');
            if (result < Int32.MinValue)
                return Int32.MinValue;
            else if (result > Int32.MaxValue)
                return Int32.MaxValue;
        }
        return (int) result;
    }
}

File: problems/0008_string-to-integer-atoi/one-loop/Solution.java
class Solution {
    public int myAtoi(String str) {
        final int sLen = str.length();
        if (sLen == 0)
            return 0;
        int i = 0;
        for (; i < sLen && str.charAt(i) == ' '; ++i);
        long sign = 1;
        if (i < sLen)
            if (str.charAt(i) == '-') {
                ++i;
                sign = -1;
            } else if (str.charAt(i) == '+') {
                ++i;
            }
        long result = 0;
        for (; i < sLen && '0' <= str.charAt(i) && str.charAt(i) <= '9'; ++i) {
            result = result * 10 + sign * (str.charAt(i) - '0');
            if (result < Integer.MIN_VALUE)
                return Integer.MIN_VALUE;
            else if (result > Integer.MAX_VALUE)
                return Integer.MAX_VALUE;
        }
        return (int) result;
    }
}
